### PR TITLE
[AN-5987] Blocked profile name and username

### PR DESCRIPTION
**Pull request by JPonte**
#### APK
[Download build #10457](http://192.168.10.18:8080/job/Pull%20Request%20Builder/10457/artifact/build/artifact/wire-dev-PR1444-10457.apk)

### CODE REVIEW

File: app/src/main/scala/com/waz/zclient/pages/main/connect/BlockedUserProfileFragment.scala [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+    var animation = super.onCreateAnimation(transit, enter, nextAnim)
+    if ((getControllerFactory.getConversationScreenController.getPopoverLaunchMode ne DialogLaunchMode.AVATAR) && (getControllerFactory.getConversationScreenController.getPopoverLaunchMode ne DialogLaunchMode.COMMON_USER)) {
+      val centerX = ContextUtils.getOrientationIndependentDisplayWidth(getActivity) / 2
+      val centerY = ContextUtils.getOrientationIndependentDisplayHeight(getActivity) / 2
+      var duration = 0
+      var delay = 0
+      // Fade out animation when starting conversation directly with this user when unblocking
+      if (!goToConversationWithUser || enter) if (nextAnim != 0) {
+        if (enter) {
+          duration = getResources.getInteger(R.integer.open_profile__animation_duration)
```
Comment (deanrobertcook) [inline app/src/main/scala/com/waz/zclient/pages/main/connect/BlockedUserProfileFragment.scala]: ContextUtils and inline

File: app/src/main/scala/com/waz/zclient/pages/main/connect/BlockedUserProfileFragment.scala [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+          delay = getResources.getInteger(R.integer.open_profile__delay)
+        }
+        else duration = getResources.getInteger(R.integer.close_profile__animation_duration)
```
Comment (deanrobertcook) [inline app/src/main/scala/com/waz/zclient/pages/main/connect/BlockedUserProfileFragment.scala]: ContextUtils and inline

File: app/src/main/scala/com/waz/zclient/pages/main/connect/BlockedUserProfileFragment.scala [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+        animation = new ProfileAnimation(enter, duration, delay, centerX, centerY)
+      }
+      else {
+        goToConversationWithUser = false
+        duration = getResources.getInteger(R.integer.framework_animation_duration_medium)
```
Comment (deanrobertcook) [inline app/src/main/scala/com/waz/zclient/pages/main/connect/BlockedUserProfileFragment.scala]: ContextUtils and inline

File: app/src/main/scala/com/waz/zclient/pages/main/connect/BlockedUserProfileFragment.scala [hunk 4]
```diff
@@ -0,0 +1,242 @@
...
+    profileImageView.setImageDrawable(profileDrawable)
+    // Split Unblock / Cancel menu when opened from group conversation
+    cancelButton.setText(getString(R.string.confirmation_menu__cancel))
+    smallUnblockButton.setText(getString(R.string.connect_request__unblock__button__text))
+    val unblockButtonColor = inject(classOf[ThemeController]).optionsDarkTheme.getTextColorPrimary
+    unblockButton.setTextColor(unblockButtonColor)
+    smallUnblockButton.setTextColor(unblockButtonColor)
+    // Hide some views irrelevant for blocking
+    footerMenu.setVisibility(View.GONE)
+    unblockButton.setVisibility(View.GONE)
```
Comment (deanrobertcook) [inline app/src/main/scala/com/waz/zclient/pages/main/connect/BlockedUserProfileFragment.scala]: Set gone in xml

File: app/src/main/scala/com/waz/zclient/pages/main/connect/BlockedUserProfileFragment.scala [hunk 5]
```diff
@@ -0,0 +1,242 @@
...
+    inflater.inflate(R.layout.fragment_blocked_user_profile, viewContainer, false)
+
+
+  override def onViewCreated(view: View, savedInstanceState: Bundle): Unit = {
+    super.onViewCreated(view, savedInstanceState)
+
+    profileImageView.setImageDrawable(profileDrawable)
+    // Split Unblock / Cancel menu when opened from group conversation
+    cancelButton.setText(getString(R.string.confirmation_menu__cancel))
+    smallUnblockButton.setText(getString(R.string.connect_request__unblock__button__text))
```
Comment (deanrobertcook) [inline app/src/main/scala/com/waz/zclient/pages/main/connect/BlockedUserProfileFragment.scala]: Set text in xml

File: app/src/main/scala/com/waz/zclient/pages/main/connect/BlockedUserProfileFragment.scala [hunk 6]
```diff
@@ -0,0 +1,242 @@
...
+    unblockButton.setTextColor(unblockButtonColor)
+    smallUnblockButton.setTextColor(unblockButtonColor)
+    // Hide some views irrelevant for blocking
+    footerMenu.setVisibility(View.GONE)
+    unblockButton.setVisibility(View.GONE)
+
+    backgroundContainer.setClickable(true)
+  }
+
+  override def onSaveInstanceState(outState: Bundle): Unit = {
```
Comment (deanrobertcook) [inline app/src/main/scala/com/waz/zclient/pages/main/connect/BlockedUserProfileFragment.scala]: obsolete as discussed

File: app/src/main/scala/com/waz/zclient/pages/main/connect/BlockedUserProfileFragment.scala [hunk 7]
```diff
@@ -0,0 +1,242 @@
...
+    outState.putString(BlockedUserProfileFragment.ARGUMENT_USER_ID, userId.currentValue.map(_.str).orNull)
+    userRequester.foreach(req => outState.putString(ARGUMENT_USER_REQUESTER, req.toString))
+    outState.putBoolean(BlockedUserProfileFragment.STATE_IS_SHOWING_FOOTER_MENU, isShowingFooterMenu)
+    super.onSaveInstanceState(outState)
+  }
+
+  private def setRegularFooterMenu(user: UserData) = {
```
Comment (deanrobertcook) [inline app/src/main/scala/com/waz/zclient/pages/main/connect/BlockedUserProfileFragment.scala]: I would inline this method, and the one from `setGroupConversationFooterMenu` then you might be able to simplify things a bit?

File: app/src/main/scala/com/waz/zclient/pages/main/connect/BlockedUserProfileFragment.scala [hunk 8]
```diff
@@ -0,0 +1,242 @@
...
+  }
+
+  private def toggleUnblockAndFooterMenu(showFooterMenu: Boolean) = {
+    if (showFooterMenu) {
+      footerMenu.setVisibility(View.VISIBLE)
+      unblockMenu.setVisibility(View.GONE)
+    } else {
+      footerMenu.setVisibility(View.GONE)
+      unblockMenu.setVisibility(View.VISIBLE)
+    }
```
Comment (deanrobertcook) [inline app/src/main/scala/com/waz/zclient/pages/main/connect/BlockedUserProfileFragment.scala]: inline if/else, maybe even use a Set?

File: app/src/main/scala/com/waz/zclient/pages/main/connect/BlockedUserProfileFragment.scala [hunk 9]
```diff
@@ -0,0 +1,242 @@
...
+    unblockMenu.setVisibility(View.GONE)
+  }
+
+  private def setGroupConversationFooterMenu(user: UserData) = {
+    unblockButton.setVisibility(View.GONE)
+    toggleUnblockAndFooterMenu(isShowingFooterMenu)
+    footerMenu.setLeftActionLabelText(getString(R.string.connect_request__footer__blocked_label))
+    footerMenu.setLeftActionText(getString(R.string.glyph__block))
+    footerMenu.setRightActionText(getString(R.string.glyph__minus))
+    footerMenu.setCallback(new FooterMenuCallback() {
```
Comment (deanrobertcook) [inline app/src/main/scala/com/waz/zclient/pages/main/connect/BlockedUserProfileFragment.scala]: you can call `footerMenu.foreach( v => ...)` and do everything in one block

File: app/src/main/scala/com/waz/zclient/pages/main/connect/BlockedUserProfileFragment.scala [hunk 10]
```diff
@@ -0,0 +1,242 @@
...
+      override def onLeftActionClicked(): Unit = toggleUnblockAndFooterMenu(false)
+      override def onRightActionClicked(): Unit = getContainer.showRemoveConfirmation(user.id)
+    })
+    cancelButton.setEnabled(true)
+    cancelButton.foreach(_.onClick(toggleUnblockAndFooterMenu(true)))
```
Comment (deanrobertcook) [inline app/src/main/scala/com/waz/zclient/pages/main/connect/BlockedUserProfileFragment.scala]: Same here

File: app/src/main/scala/com/waz/zclient/pages/main/connect/BlockedUserProfileFragment.scala [hunk 11]
```diff
@@ -0,0 +1,242 @@
...
+    smallUnblockButton.setEnabled(true)
+    smallUnblockButton.foreach(_.onClick(unblockUser(user)))
```
Comment (deanrobertcook) [inline app/src/main/scala/com/waz/zclient/pages/main/connect/BlockedUserProfileFragment.scala]: same here

File: app/src/main/scala/com/waz/zclient/pages/main/connect/BlockedUserProfileFragment.scala [hunk 12]
```diff
@@ -0,0 +1,242 @@
...
+  private lazy val userNameView = view[TypefaceTextView](R.id.user_name)
+  private lazy val userUsernameView = view[TypefaceTextView](R.id.user_handle)
+
+  override def onCreate(savedInstanceState: Bundle): Unit = {
+    super.onCreate(savedInstanceState)
+
+    if (savedInstanceState != null) {
+      userId ! UserId(savedInstanceState.getString(BlockedUserProfileFragment.ARGUMENT_USER_ID))
+      userRequester = Option(IConnectStore.UserRequester.valueOf(savedInstanceState.getString(ARGUMENT_USER_REQUESTER)))
+      isShowingFooterMenu = savedInstanceState.getBoolean(BlockedUserProfileFragment.STATE_IS_SHOWING_FOOTER_MENU)
```
Comment (deanrobertcook) [inline app/src/main/scala/com/waz/zclient/pages/main/connect/BlockedUserProfileFragment.scala]: as discussed, these can be lazy vals

**Comment by someoneEsle** [comment]
- [ ] Bug 1
Same bug affects people you've sent a connection request to

**Comment by someoneEsle** [comment]
- [ ] Bug 2
and people who sent you a connection request

**Comment by red-avtovo** [comment]
build again

**Comment by someoneEsle** [comment]
- [ ] Bug 3
The app crashes consistently when trying to reproduce bug 2

**Comment by someoneEsle** [comment]
- [ ] Bug 4
1. Tap a participant you sent a connection request to
2. go back
3. Tap a participant who sent you a connection request
4. go back
5. Tap the participant from step 1 --> there is no "cancel request" button

**Comment by JPonte** [comment]
build again

**Comment by JPonte** [comment]
build again

**Comment by JPonte** [comment]
build again

**Comment by JPonte** [comment]
build again

**Pull request merged**